iModelCore/ECDb/Tests/NonPublished/ClassViewsTests.cpp

```diff
@@ -0,0 +1,631 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
I'd like to see tests for:

- Return data and cover all primitive types.
- structs and arrays (should fail as they are not supported)
- Navigation props (are they supported)
- Are any implicit conversions supported if there is a mismatch between view property and selected property?
- Relationship view class
- Metadata mismatch like different persistence units
- instance query (is it supported?)

Reply:
Sure, anything you can do with ECSQL SELECT can be done in View.  View Query is substituted for View Class in query and all type resolutions will happen as in any sub query. We need tests for all above 

My only concerns are edge cases e.g. Conflict between an "Alias" used in "View sub query" that is also used in "query that references the view" and the user might get an error where he does not see a problem with the query but its failing when the view is substituted. To sandbox the view query, we need to change all the aliases used in the subquery to add prefixes, so they do not collide with the primary query.

Reply:
@ColinKerr we have tests for all of the above except for this type:
- Metadata mismatch like different persistence units
What did you have in mind? The view query does not need to have a KindOfQuantity, so this will always use the one specified on the property. Is that a concern?

Reply:
According to discussion this limitation will be mentioned in the final documentation in the itwinjs-core repo